### PR TITLE
feature/home_uistate

### PR DESCRIPTION
**Pull request by MoatazBadawy**

### CODE REVIEW

File: app/src/main/java/com/moataz/afternoonhadeeth/domain/usecase/home/GetHadithEveryDayUseCase.kt [hunk 1]
```diff
@@ -17,31 +17,31 @@ class GetHadithEveryDayUseCase @Inject constructor(
     private val repository: AfternoonHadithRepositoryImpl,
     @ApplicationContext private val appContext: Context,
```
Comment (MahmoudMabrok) [inline app/src/main/java/com/moataz/afternoonhadeeth/domain/usecase/home/GetHadithEveryDayUseCase.kt:18]: Use case should depend on interface not implemnetation

Comment (MahmoudMabrok) [reply app/src/main/java/com/moataz/afternoonhadeeth/domain/usecase/home/GetHadithEveryDayUseCase.kt:18]: do we need _**context**_ here?

Comment (MoatazBadawy) [reply app/src/main/java/com/moataz/afternoonhadeeth/domain/usecase/home/GetHadithEveryDayUseCase.kt:18]: Done 👍

Comment (MoatazBadawy) [reply app/src/main/java/com/moataz/afternoonhadeeth/domain/usecase/home/GetHadithEveryDayUseCase.kt:18]: I have used context to provide it to sharedprefs to save some values, but now I have replaced it with datastore.

**Pull request merged**